trainer/callbacks.py

```diff
@@ -59,6 +59,15 @@ def on_epoch_end(trainer) -> None:
         if hasattr(trainer.optimizer, "on_epoch_end"):
             trainer.optimizer.on_epoch_end(trainer)
 
+    @staticmethod
+    def before_main_optimizer_step(trainer, loss_dict, optimizer):
```

Comment:
why do you need to pass loss_dict and optimizer? They are already reachable by the trainer.

Reply:
I need to have access to them in the model method --> https://github.com/coqui-ai/TTS/blob/1afd685d230a70b9651d67392b5096dd76042ceb/TTS/tts/models/tacotron2.py#L303